tiledb/sm/filesystem/s3_thread_pool_executor.cc

```diff
@@ -50,6 +50,7 @@ S3ThreadPoolExecutor::S3ThreadPoolExecutor(ThreadPool* const thread_pool)
 }
 
 S3ThreadPoolExecutor::~S3ThreadPoolExecutor() {
+  throw_if_not_ok(Stop());
```

Comment:
Can you make `Stop` return void? It never returns a faulted status so it should be easy.